README.md

```diff
@@ -3,16 +3,133 @@ Dagger Reflect
+ * `FooComponent foo = DaggerFooComponent.create()`  
+   &rarr; `FooComponent foo = Dagger.create(FooComponent.class)`
+ * `FooComponent foo = DaggerFooComponent.builder().….build()`  
+   &rarr; `FooComponent foo = Dagger.builder(FooComponent.Builder.class).….build()`
+
+For the release build using `dagger-codegen` and shirking you'll need to add this to the configuration:
+```proguard
+# [dagger-reflect] Make sure to keep entry points that dagger.Dagger reflectively accesses
+# fixes: Unable to find generated component implementation com.example.Daggera for component com.example.a
+
+# Keep annotation to be able to match it below
+-keep class dagger.Component
+# Keep the names of the Component interfaces so we can prefix them with "Dagger"
+-keepnames @dagger.Component interface **
+# Keep the names of the generated Component implementations so the prefixed lookup succeeds
+-keepnames class ** implements @dagger.Component **
+# Keep the builder() method in the generated Component to reflectively call it
+-keepclassmembers class ** implements @dagger.Component ** {
+	public static ** builder();
+}
+# Keep the create() method in the generated Component to reflectively call it
+-keepclassmembers class ** implements @dagger.Component ** {
+	public static <2> create();
+}
+```
+
+### SNAPSHOT
 
 Snapshots of the development version are available in [Sonatype's `snapshots` repository][snap].
+Don't forget to suffix `VERSION_DAGGER_REFLECT` above with `-SNAPSHOT`, the rest of the usage is the same.
 
+```gradle
+repositories {
+  maven { name = "Sonatype SNAPSHOTs"; url = "https://oss.sonatype.org/content/repositories/snapshots/" }
+}
+```
+
+
+Limitations
+-----------
+
+ * `@Component`s and `@Component.Builder`s must be interfaces  
+   While Dagger allows `abstract class`es for these, `dagger-reflect` uses Java Proxies to handle invocations which cannot extend classes at runtime.
+ 
+ * `@Component` interfaces must be public (default visible won't work):  
+   This is due to a limitation in Java, where instances of proxies cannot create another proxy instance where the second interface is not public. This prevents proxies of builders from creating proxies of the component. See `dagger.reflect.ComponentBuilderInvocationHandler.create`.
+ 
+ * There are some missing features that are not yet implemented.  
+   They can be found by looking at [`notImplemented` calls in the code][notImplemented].  
+   (Beware: GitHub shows only the first few occurrences in each file, not all of them.)
+
+Compatibility
+-------------
+
+ * ProGuard/DexGuard/R8: since the dependency injection entry point (e.g. `@Component.Builder`) is being reflected with either usage case, you'll lose some shrinkability, but the majority of the generated `@Component` code using `@Module`s will be shrinked the same as before.
+
+ * `dagger-android`: it uses a lot of Dagger features, namingly:
```

Comment:
Most of these are implementation details. I don't think it's our responsibility or useful to document all this.